packages/test/test-gc-sweep-tests/src/dataObjectWithChildDataObject.ts

```diff
@@ -25,40 +30,48 @@ export class RootDataObjectWithChildDataObject extends DataObjectWithCounter {
 
     protected async hasInitialized(): Promise<void> {
         await super.hasInitialized();
+        this.uniqueId = makeRandom().uuid4();
         await this.createAndReferenceChild();
-        // TODO: Start logic goes here after we agree on what it should be.
+        this.start();
     }
 
-    // TODO: Make private - left public so the build passes
-    public async unreferenceChild() {
+    private unreferenceChild() {
         assert(this.child !== undefined, `Child should be defined when unreferencing!`);
-        this.unreferencedChild = this.child;
         this.child.stop();
         this.root.delete(this.childKey);
-        this.unrefTimeStampMs = Date.now();
     }
 
     private async createAndReferenceChild() {
         this.child = await dataObjectWithCounterFactory.createInstance(this.context.containerRuntime);
```

Comment:
`assert(this.child === undefined)` first to avoid inadvertently leaving an unreferenced child running?